### DISCUSSION TITLE
Multi-query support

### DISCUSSION THREAD

**Discussion by tyzbit** [marcusolsson/grafana-hexmap-panel#20] [Ideas]
Separately, each of these queries work fine to color or size the nodes, but when providing these queries together, the second query doesn't size the node, nor does the value show in the labels. I am new to grafana so I definitely could be doing something wrong. It seems like the values returned by the query aren't the same type (int, float) as the first query and as a result it fails.

### Query 1
```
avg(container_memory_usage_bytes{container!="POD",namespace=~"$namespace",node=~"$node"} / 1024 / 1024) by (pod,container,node)
```

### Query 2
```
avg(sqrt(sqrt(sqrt(sqrt(container_memory_usage_bytes{container!="POD",namespace=~"$namespace",node=~"$node"} / 1024 / 1024))))) by (pod,container,node)
```

**Comment by marcusolsson** [comment]
Unfortunately, the plugin is limited to a single query right now. Out of curiosity, how would you expect multiple queries to be visualized?

**Comment by tyzbit** [comment]
Each query should return no more than one numeric value column, or more specifically you can only use up to two value columns from the combined queries. The plugin would then run statistics on the values for each query to determine min, max, avg etc. Then you could choose a value column to color **or** size the nodes. Then one other value column could be used to do the remaining transformation type on the nodes. I don't think it would work with more than 2 distinct values since there are only two facets of the nodes that can be changed (size, color). Perhaps you could group by one more value by creating buckets based on the value (in logarithmic steps, in user-defined groups, in percentiles).

My goal with these queries specifically was to let each container be a node, then color the node corresponding to the relative CPU usage of the container. Finally, let the size of the node correspond to the relative memory usage for the container.
